conjure-error/src/lib.rs

```diff
@@ -57,102 +57,40 @@ impl ErrorCode {
 /// A trait implemented by Conjure error types.
 pub trait ErrorType {
     /// Returns the error's code.
-    fn code(&self) -> ErrorCode;
+    fn code() -> ErrorCode;
 
     /// Returns the error's name.
     ///
     /// The name must be formatted like `NamespaceName:ErrorName`.
-    fn name(&self) -> &str;
-
-    /// Returns the error's instance ID, if it stores one.
-    ///
-    /// Conjure-generated error types return `None`, but other implementations like those for `SerializableError`
-    /// and `WithInstanceId` return a value.
-    fn instance_id(&self) -> Option<Uuid>;
```

Comment:
We do need the ability to manually set the instance ID for https://github.com/palantir/conjure-rust-runtime/pull/235, but this old method still doesn't feel like the right place to put it...